core/service/src/consensus.rs

```diff
@@ -0,0 +1,286 @@
+// Copyright 2018 Parity Technologies (UK) Ltd.
```

Comment:
Why exists all this code 2 times?

Reply:
it only exists once: here. the other version isn't actually working. but I'll add a ticket to clarify which part of that is generic and clean that up.